plugin/selection_range.py

```diff
@@ -0,0 +1,55 @@
+from .core.protocol import Range
+from .core.protocol import Request
+from .core.registry import get_position
+from .core.registry import LspTextCommand
+from .core.sessions import method_to_capability
+from .core.typing import Any, Dict, Optional, List
+from .core.views import range_to_region
+from .core.views import selection_range_params
+import sublime
+
+
+class LspExpandSelectionCommand(LspTextCommand):
+    method = 'textDocument/selectionRange'
+    capability = method_to_capability(method)[0]
+
+    def __init__(self, view: sublime.View) -> None:
+        super().__init__(view)
+        self._regions = []  # type: List[sublime.Region]
+        self._change_count = 0
+
+    def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
+        session = self.session(self.capability, get_position(self.view, event))
+        if session:
+            params = selection_range_params(self.view)
+            self._regions.extend(self.view.sel())
+            self._change_count = self.view.change_count()
+            session.send_request(Request(self.method, params), self.on_result, self.on_error)
+
+    def on_result(self, params: Any) -> None:
+        if self._change_count != self.view.change_count():
+            return
+        if params:
+            regions = [self._smallest_containing(region, param) for region, param in zip(self._regions, params)]
+        else:
+            regions = self._regions
+            window = self.view.window()
+            if window:
```

Comment:
What's the reason for setting status on the window and not view? This is a view command after all.

Also, is that message ever cleared?
I guess it might be cleared by "something else" eventually but should we rely on that?

I guess that's another reason to use view status as then we know our key and can clear it reliably.

Reply:
The `window.status_message` method also clears the message after a short while.

Reply:
Should I use the same for CAOS then, to be consistent? I admit it shows in a better place then view status which tends to be crammed on the left and shift everything.